src/main/java/com/refinedmods/refinedstorage/apiimpl/autocrafting/task/v6/IoUtil.java

```diff
@@ -0,0 +1,187 @@
+package com.refinedmods.refinedstorage.apiimpl.autocrafting.task.v6;
+
+import com.refinedmods.refinedstorage.api.network.INetwork;
+import com.refinedmods.refinedstorage.api.storage.disk.IStorageDisk;
+import com.refinedmods.refinedstorage.api.util.Action;
+import com.refinedmods.refinedstorage.api.util.IComparer;
+import com.refinedmods.refinedstorage.api.util.IStackList;
+import com.refinedmods.refinedstorage.api.util.StackListEntry;
+import com.refinedmods.refinedstorage.apiimpl.API;
+import net.minecraft.item.ItemStack;
+import net.minecraftforge.fluids.FluidStack;
+import net.minecraftforge.fluids.capability.IFluidHandler;
+import net.minecraftforge.items.IItemHandler;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+import javax.annotation.Nullable;
+import java.util.*;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+
+public final class IoUtil {
+    private static final Logger LOGGER = LogManager.getLogger(IoUtil.class);
+
+    private static final int DEFAULT_EXTRACT_FLAGS = IComparer.COMPARE_NBT;
+
+    public static IStackList<ItemStack> extractFromInternalItemStorage(IStackList<ItemStack> list, IStorageDisk<ItemStack> storage, Action action) {
```

Comment:
1st parameter is a Collection

Reply:
I changed it so it takes a stacklist, since it was using `Collection<StackListEntry<?>>`, so if it enforces a stack list entry might as well just use the stacklist. Makes the signature easier to read.

Reply:
ah. Then you need to remove getStacks() from where this gets called though.

Reply:
That is already done?

Reply:
Then you did not commit those changes yet.